src/components/App.tsx

```diff
@@ -20,39 +24,103 @@ class App extends React.Component<Record<string, unknown>, AppState> {
     super(props);
     this.state = {
       isVerified: false,
+      userReserves: [],
     };
   }
 
   private async setUpWeb3() {
     this.web3 = await getWeb3();
     this.mainAccount = (await this.web3.eth.getAccounts())[0];
-    console.log(this.mainAccount);
+    console.log("MAIN ACCOUNT: ", this.mainAccount);
     this.setState({
       isVerified: true,
     });
   }
 
   componentDidMount() {
     this.setUpWeb3();
+    // TODO: fetch real ETH price
+    const ethPriceUsd = 2700;
+    // TODO: only run after this.mainAccount is fetched - at this point it's null
```

Comment:
@adotcruz what would be a good spot for making a call that requires `this.mainAccount` to be non-null? I tried in render but it's not guaranteed

Reply:
just place it in `this.setUpWeb3` the issue with your logic here is `componentDidMount` does not wait on the promise to resolve. Even though `this.setUpWeb3` is async, only the function logic is synced up, this call here works as a fire and forget, `componentDidMount` will resolve before `this.setUpWeb3` resolves.

Lmk if this makes sense, I can send you some documentation on async/await when used in a non async function.

Reply:
Makes sense that `componentDidMount` works like a fire and forget
Moved to `setUpWeb3` and renamed to `getAccountInfo`